### PR TITLE
Feature/improve icon names

### PR DESCRIPTION
**Pull request by aappoalander**
## Description

Exported icon names are aligned with used design terminology: baseIcons, illustrativeIcons, doctypeIcons and componentIcons.

This PR splits original SuomifiIcon to SuomifiIcon (baseIcons), SuomifiStaticIcon (Illustrative and doctypeIcons and SuomifiComponentIcon (componentIcons) to allow overlapping icon names for differrent icon types. Icons used internally in suomifi-ui-components are moved to SuomifiComponentIcon.

Folder and file structures are changed to reflect these changes.

## Motivation and Context

Currently different types of icons use related prefix like staticIllustration to indicate the icon type and prevent overlapping names. This makes the icons cumbersome to use and difficult to read the icon names. In addition the API allows passing fill color for illustrative and doctype icons as well even though it's not supported. Moving component icons out of static icons set allows removing exports for these icons from suomifi-ui-components easily.

## How Has This Been Tested?

Tested with Suomi.fi Design System Site locally using Verdaccio and the icon component exported by suomifi-icons (instead of the one from ui-components that is used normally).

### CODE REVIEW

**Review by ketsappi — APPROVED** [review]
Looking good! Didn't found any problems or typos.  

Tested locally with CRA TS-project and using the `SuomifiIcon, SuomifiComponentIcon, 
SuomifiIllustrativeIcon` classes with some random icons from there.

**Review by LJKaski — COMMENTED** [review]
Seems good to me. The only thing worth mentioning I found was the switch from american to UK english in the token name 'authorizations' --> 'authorisation'. Is that on purpose and is that consistent throughout the services (no a huge issue in any case)?

**Comment by aappoalander** [comment]
> Seems good to me. The only thing worth mentioning I found was the switch from american to UK english in the token name 'authorizations' --> 'authorisation'. Is that on purpose and is that consistent throughout the services (no a huge issue in any case)?

@LJKaski this is as defined in confluence. The change was made to align with the style used with rest of the icon names.

**Review by ketsappi — APPROVED** [review]

**Review by LJKaski — COMMENTED** [review]
The fill and color props don't seem to do anything on the toggle ComponentIcon.

**Comment by aappoalander** [comment]
> The fill and color props don't seem to do anything on the toggle ComponentIcon.

This is actually desired behaviour. The only icon in component icons currently has internal class handles to apply colors to svg parts. This might not be the case for future component icons and we might want to use fill/color with those. Fill and color prop can both be applied even if not explicitly defined in the API since we allow custom props.

Will remove explicit definition and handling of fill and color props for component icon.

File: src/index.tsx [hunk 1]
```diff
@@ -1,52 +1,100 @@
...
+interface SuomifiIconBaseInterface<T> {
+  /** Icon name */
+  icon: T;
+  /** Custom classname to extend or customize */
   className?: string;
   // Allow passing unsupported custom props to SVG without providing an API
   [key: string]: any;
 }
 
+interface FillPropsInterface {
```
Comment (ketsappi) [inline src/index.tsx]: Not really a problem, but maybe a useful convention that the interfaces should be introduced before they are used in the file.

**Review by ketsappi — APPROVED** [review]

**Review by LJKaski — APPROVED** [review]

File: src/index.tsx (hunk 1 above)
Comment (aappoalander) [reply src/index.tsx]: Agreed, fixed this. We should have linter rules in place to take care of this. Maybe even use the same linter settings for all repos.

**Pull request merged**